### PR TITLE
vectorization improvements for TPS transform(), apply(), and inverse()

### PR DESCRIPTION
**Pull request by djkapner**

### CODE REVIEW

File: renderapi/transform/leaf/thin_plate_spline.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from renderapi.errors import RenderError, EstimationError
 from renderapi.utils import encodeBase64, decodeBase64
 from .transform import Transform
+import scipy.spatial
+import scipy.special
```
Comment (fcollman) [inline renderapi/transform/leaf/thin_plate_spline.py]: duplicate line

Comment (djkapner) [reply renderapi/transform/leaf/thin_plate_spline.py]: it is not duplicate, but I ended up not using scipy.special ... would have thought flake8 would have told me that... will remove

**Comment by djkapner** [comment]
let's not merge this just yet. I found a scipy thin plate radial basis function that is even faster...

**Comment by fcollman** [comment]
@RussTorres i'm good with merging this are you?

**Comment by RussTorres** [comment]
I mentioned this before, but not as a review -- this PR makes scipy an actual dependency.  If we're okay with that (and I think we are) we should add it to the requirements file.

**Comment by djkapner** [comment]
I will make that change in the PR once Forrest confirms he is ok with the dependency

**Comment by djkapner** [comment]
there do appear to be a couple of numpy only ideas instead of scipy.spatial.cdist:
https://stackoverflow.com/questions/52030458/vectorized-spatial-distance-in-python-using-numpy

**Pull request merged**